library/std/src/os/windows/io/handle.rs

```diff
@@ -157,24 +158,40 @@ impl BorrowedHandle<'_> {
     }
 }
 
-#[stable(feature = "io_safety", since = "1.63.0")]
-impl TryFrom<HandleOrNull> for OwnedHandle {
-    type Error = NullHandleError;
+macro_rules! impl_handle_or {
```

Comment:
Especially when combined with the suggestion above, I don't think that a macro is really necessary here.

Reply:
I would likely still use a macro due to the shared code between the implementations, but I don't have strong opinions if the preference is to avoid macros here.